### ISSUE TITLE
How do I suppress debug logging?

### ISSUE THREAD

**Issue by minecraft2048** [FabriceSalvaire/PySpice#121]
### Expected Behaviour
There is no debug output
### Actual Behaviour
There is a lot of debug output, which is space consuming if you use Jupyter
```
2018-06-10 08:48:59,781 - PySpice.Spice.Netlist.__new__ - DEBUG - Element don't define a __pins__ attribute
2018-06-10 08:48:59,782 - PySpice.Spice.Netlist.__new__ - DEBUG - FixedPinElement don't define a __pins__ attribute
2018-06-10 08:48:59,783 - PySpice.Spice.Netlist.__new__ - DEBUG - SubCircuitElement don't define a __pins__ attribute
2018-06-10 08:48:59,785 - PySpice.Spice.Netlist.__new__ - DEBUG - Resistor don't define a __pins__ attribute
2018-06-10 08:48:59,786 - PySpice.Spice.Netlist.__new__ - DEBUG - SemiconductorResistor don't define a __pins__ attribute
2018-06-10 08:48:59,787 - PySpice.Spice.Netlist.__new__ - DEBUG - BehavioralResistor don't define a __pins__ attribute
2018-06-10 08:48:59,788 - PySpice.Spice.Netlist.__new__ - DEBUG - Capacitor don't define a __pins__ attribute
2018-06-10 08:48:59,789 - PySpice.Spice.Netlist.__new__ - DEBUG - SemiconductorCapacitor don't define a __pins__ attribute
2018-06-10 08:48:59,789 - PySpice.Spice.Netlist.__new__ - DEBUG - BehavioralCapacitor don't define a __pins__ attribute
2018-06-10 08:48:59,790 - PySpice.Spice.Netlist.__new__ - DEBUG - Inductor don't define a __pins__ attribute
2018-06-10 08:48:59,791 - PySpice.Spice.Netlist.__new__ - DEBUG - BehavioralInductor don't define a __pins__ attribute
2018-06-10 08:48:59,792 - PySpice.Spice.Netlist.__new__ - DEBUG - CoupledInductor don't define a __pins__ attribute
2018-06-10 08:48:59,792 - PySpice.Spice.Netlist.__new__ - DEBUG - VoltageControlledSwitch don't define a __pins__ attribute
2018-06-10 08:48:59,793 - PySpice.Spice.Netlist.__new__ - DEBUG - CurrentControlledSwitch don't define a __pins__ attribute
2018-06-10 08:48:59,794 - PySpice.Spice.Netlist.__new__ - DEBUG - VoltageSource don't define a __pins__ attribute
2018-06-10 08:48:59,795 - PySpice.Spice.Netlist.__new__ - DEBUG - CurrentSource don't define a __pins__ attribute
2018-06-10 08:48:59,795 - PySpice.Spice.Netlist.__new__ - DEBUG - VoltageControlledCurrentSource don't define a __pins__ attribute
2018-06-10 08:48:59,796 - PySpice.Spice.Netlist.__new__ - DEBUG - VoltageControlledVoltageSource don't define a __pins__ attribute
2018-06-10 08:48:59,797 - PySpice.Spice.Netlist.__new__ - DEBUG - CurrentControlledCurrentSource don't define a __pins__ attribute
2018-06-10 08:48:59,797 - PySpice.Spice.Netlist.__new__ - DEBUG - CurrentControlledVoltageSource don't define a __pins__ attribute
2018-06-10 08:48:59,798 - PySpice.Spice.Netlist.__new__ - DEBUG - BehavioralSource don't define a __pins__ attribute
2018-06-10 08:48:59,799 - PySpice.Spice.Netlist.__new__ - DEBUG - NonLinearVoltageSource don't define a __pins__ attribute
2018-06-10 08:48:59,800 - PySpice.Spice.Netlist.__new__ - DEBUG - NonLinearCurrentSource don't define a __pins__ attribute
2018-06-10 08:48:59,801 - PySpice.Spice.Netlist.__new__ - DEBUG - JunctionFieldEffectTransistor don't define a __pins__ attribute
2018-06-10 08:48:59,801 - PySpice.Spice.Netlist.__new__ - DEBUG - Mesfet don't define a __pins__ attribute
2018-06-10 08:48:59,802 - PySpice.Spice.Netlist.__new__ - DEBUG - LosslessTransmissionLine don't define a __pins__ attribute
2018-06-10 08:48:59,804 - PySpice.Spice.Netlist.__new__ - DEBUG - LossyTransmission don't define a __pins__ attribute
2018-06-10 08:48:59,804 - PySpice.Spice.Netlist.__new__ - DEBUG - CoupledMulticonductorLine don't define a __pins__ attribute
2018-06-10 08:48:59,805 - PySpice.Spice.Netlist.__new__ - DEBUG - SingleLossyTransmissionLine don't define a __pins__ attribute
2018-06-10 08:48:59,806 - PySpice.Spice.Netlist.__new__ - DEBUG - XSpiceElement don't define a __pins__ attribute
2018-06-10 08:48:59,806 - PySpice.Spice.Netlist.__new__ - DEBUG - GSSElement don't define a __pins__ attribute
2018-06-10 08:48:59,808 - PySpice.Spice.Netlist.__new__ - DEBUG - SinusoidalVoltageSource don't define a __pins__ attribute
2018-06-10 08:48:59,809 - PySpice.Spice.Netlist.__new__ - DEBUG - SinusoidalCurrentSource don't define a __pins__ attribute
2018-06-10 08:48:59,809 - PySpice.Spice.Netlist.__new__ - DEBUG - AcLine don't define a __pins__ attribute
2018-06-10 08:48:59,810 - PySpice.Spice.Netlist.__new__ - DEBUG - PulseVoltageSource don't define a __pins__ attribute
2018-06-10 08:48:59,811 - PySpice.Spice.Netlist.__new__ - DEBUG - PulseCurrentSource don't define a __pins__ attribute
2018-06-10 08:48:59,812 - PySpice.Spice.Netlist.__new__ - DEBUG - ExponentialVoltageSource don't define a __pins__ attribute
2018-06-10 08:48:59,812 - PySpice.Spice.Netlist.__new__ - DEBUG - ExponentialCurrentSource don't define a __pins__ attribute
2018-06-10 08:48:59,813 - PySpice.Spice.Netlist.__new__ - DEBUG - PieceWiseLinearVoltageSource don't define a __pins__ attribute
2018-06-10 08:48:59,814 - PySpice.Spice.Netlist.__new__ - DEBUG - PieceWiseLinearCurrentSource don't define a __pins__ attribute
2018-06-10 08:48:59,815 - PySpice.Spice.Netlist.__new__ - DEBUG - SingleFrequencyFMVoltageSource don't define a __pins__ attribute
2018-06-10 08:48:59,816 - PySpice.Spice.Netlist.__new__ - DEBUG - SingleFrequencyFMCurrentSource don't define a __pins__ attribute
2018-06-10 08:48:59,817 - PySpice.Spice.Netlist.__new__ - DEBUG - AmplitudeModulatedVoltageSource don't define a __pins__ attribute
2018-06-10 08:48:59,818 - PySpice.Spice.Netlist.__new__ - DEBUG - AmplitudeModulatedCurrentSource don't define a __pins__ attribute
2018-06-10 08:48:59,819 - PySpice.Spice.Netlist.__new__ - DEBUG - RandomVoltageSource don't define a __pins__ attribute
2018-06-10 08:48:59,820 - PySpice.Spice.Netlist.__new__ - DEBUG - RandomCurrentSource don't define a __pins__ attribute
```

How can I remove all of this? And if its possible, how can I remove all the debug log without removing spice simulator output, like this:
```
2018-06-10 00:03:12,345 - PySpice.Spice.Xyce.Server.XyceServer._parse_stdout - DEBUG - 

*****
***** Welcome to the Xyce(TM) Parallel Electronic Simulator
*****
***** This is version Xyce Release 6.8-opensource
***** Date: Sun Jun 10 00:03:12 AEST 2018


***** Executing netlist /tmp/tmp6jh9c_p7/input.cir

***** Reading and parsing netlist...
Netlist warning: No print specified
***** Setting up topology...

***** Device Count Summary ...
       R level 1 (Resistor)                   2
       V level 1 (Independent Voltage Source) 1
       ----------------------------------------
       Total Devices                          3
***** Setting up matrix structure...
***** Number of Unknowns = 3
***** Initializing...

------------------------------------------------------------
Operating point information:
Number of Independent Voltage Source models: 1
          name V level 1 (Independent Voltage Source)
          type                
         level               1

Number of Independent Voltage Source instances: 1
          name          VINPUT
         model V level 1 (Independent Voltage Source)
            NUM              0
   DCSOURCETYPE              6
TRANSIENTSOURCETYPE              6
              V              0
              T              0
             FS              0
        ACPHASE              0
          ACMAG              0
             TR              0
             SF              0
             V0              0
             TD              0
            TD2              0
             V1              0
   ACSOURCETYPE              7
           DCV0             10
             TF              0
              R              0
             PW              0
             FC              0
             V2              0
            PER              0
             VA              0
           FREQ              0
          THETA              0
            MDI              0
           TAU2              0
          PHASE              0
            TD1              0
           TAU1              0


Number of Resistor models: 1
          name R level 1 (Resistor)
          type                
         level               1
            TC2              0
            RSH              0
           TNOM             25
            TC1              0
              R              1
           DEFW          1e-05
         NARROW              0

Number of Resistor instances: 2
          name              R1             R2
         model R level 1 (Resistor)R level 1 (Resistor)
            TC2              0              0
              M              1              1
              L              0              0
          DTEMP              0              0
              W          1e-05          1e-05
           TEMP             25             25
              R           9000           1000
            TC1              0              0

------------------------------------------------------------

***** Solution Summary *****
	Number Successful Steps Taken:		1
	Number Failed Steps Attempted:		0
	Number Jacobians Evaluated:		2
	Number Iteration Matrix Factorizations:	0
	Number Linear Solves:			2
	Number Failed Linear Solves:		0
	Number Residual Evaluations:		3
	Number Nonlinear Convergence Failures:	0
	Total Residual Load Time:		1.50204e-05 seconds
	Total Jacobian Load Time:		5.24521e-06 seconds
	Total Linear Solution Time:		9.82285e-05 seconds


***** Total Simulation Solvers Run Time: 0.000856876 seconds
***** Total Elapsed Run Time:            0.00299692 seconds
*****
***** End of Xyce(TM) Simulation
*****

Timing summary of 1 processor
                 Stats                   Count       CPU Time              Wall Time
---------------------------------------- ----- --------------------- ---------------------
Xyce                                         1        0.010 (100.0%)        0.010 (100.0%)
  Analysis                                   1        0.001 ( 8.43%)        0.001 ( 7.88%)
    DC Sweep                                 1        0.001 ( 8.13%)        0.001 ( 7.59%)
      Solve                                  1        0.000 ( 4.48%)        0.000 ( 4.19%)
        Residual                             3        0.000 ( 0.22%)        0.000 ( 0.19%)
        Jacobian                             2        0.000 ( 0.08%)        0.000 ( 0.08%)
        Linear Solve                         2        0.000 ( 1.05%)        0.000 ( 0.99%)
      Successful Step                        1        0.000 ( 3.26%)        0.000 ( 3.06%)
  Netlist Import                             1        0.000 ( 4.82%)        0.001 ( 5.86%)
    Parse Context                            1        0.000 ( 1.03%)        0.000 ( 1.92%)
    Distribute Devices                       1        0.000 ( 2.23%)        0.000 ( 2.21%)
    Verify Devices                           1        0.000 ( 0.02%)        0.000 ( 0.02%)
    Instantiate                              1        0.000 ( 0.93%)        0.000 ( 0.88%)
  Late Initialization                        1        0.001 ( 7.83%)        0.001 ( 7.33%)
    Global Indices                           1        0.000 ( 0.35%)        0.000 ( 0.33%)
  Setup Matrix Structure                     1        0.000 ( 1.61%)        0.000 ( 1.52%)
```

**Comment by FabriceSalvaire** [comment]
look in PySpice/Config/logging.yml
also you can change some _logger.debug() to info

**Comment by FabriceSalvaire** [comment]
According to https://docs.python.org/3/howto/logging.html#custom-levels it could be a bad idea, but I agree we need a simple mechanism to switch on/off spice logging.

Anyway PySpice logging must be cleaned:
- set info level by default
- check the logging level for each line of codes

**Comment by FabriceSalvaire** [comment]
partially fixed in 29cb9af

**Comment by FabriceSalvaire** [comment]
fix in c68eee9a8cd1d4292f9d369c01506b3858359a69

**Comment by FabriceSalvaire** [comment]
Should be outdated now
